plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoPrivileges.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mongodb;
+
+import com.google.common.collect.ImmutableMap;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import org.bson.Document;
+import org.testng.annotations.Test;
+
+import java.util.Locale;
+import java.util.Optional;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.TEST_COLLECTION;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.TEST_DATABASE;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.createTestRole;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.createTestUser;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestMongoPrivileges
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        AuthenticatedMongoServer mongoServer = closeAfterClass(setupMongoServer());
+        return createMongoQueryRunner(mongoServer.testUserConnectionString().getConnectionString());
+    }
+
+    @Test
+    public void testSchemasVisibility()
+    {
+        assertThat(getQueryRunner().execute("SHOW SCHEMAS FROM mongodb")
```

Comment:
This test succeeds even before the change. Is it possible to make the test that failed previously?

Reply:
Unfortunately I'm not able to configure MongoDB access control in such way to simulate case from here: https://www.mongodb.com/docs/manual/reference/command/listDatabases/#behavior for `MongoDB 4.0.6+`. I use `4.2.0`
I would still leave this test but as separate commit - I will extract this